core/src/main/java/aima/core/environment/support/BidirectionalRomaniaProblem.java

```diff
@@ -0,0 +1,118 @@
+package aima.core.environment.support;
+
+import aima.core.environment.map2d.GoAction;
+import aima.core.environment.map2d.InState;
+import aima.core.environment.map2d.Map2DFunctionFactory;
+import aima.core.environment.map2d.SimplifiedRoadMapOfPartOfRomania;
+import aima.core.search.api.ActionsFunction;
+import aima.core.search.api.BidirectionalProblem;
+import aima.core.search.api.Problem;
+import aima.core.search.api.ResultFunction;
+import aima.core.search.api.StepCostFunction;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * @author wormi
+ */
+public class BidirectionalRomaniaProblem implements BidirectionalProblem<GoAction, InState> {
+
+  private final SimplifiedRoadMapOfPartOfRomania map = new SimplifiedRoadMapOfPartOfRomania();
+  private final String initialLocation;
+  private final String[] goalLocations;
+  private final ResultFunction<GoAction, InState> resultFunction;
+
+  private final ActionsFunction<GoAction, InState> actionsFunction;
+  private final StepCostFunction<GoAction, InState> stepCostFunction;
+
+  {
+    actionsFunction = Map2DFunctionFactory.getActionsFunction(map);
+    resultFunction = Map2DFunctionFactory.getResultFunction(map);
+    stepCostFunction = Map2DFunctionFactory.getStepCostFunction(map);
+  }
+
+  BidirectionalRomaniaProblem(String initialLocation, String... goalLocations) {
+    this.initialLocation = initialLocation;
+    this.goalLocations = goalLocations;
+  }
+
+  @Override
+  public Problem<GoAction, InState> getOriginalProblem() {
+    return new MyOriginalProblem();
+  }
+
+  @Override
+  public List<Problem<GoAction, InState>> getReverseProblems() {
+    List<Problem<GoAction, InState>> problems = new LinkedList<>();
+    for (String goalLocation : goalLocations) {
+      problems.add(new MyReverseProblem(goalLocation));
+    }
+    return problems;
+  }
+
+
```

Comment:
I am not sure why you need these two custom classes below (MyOriginalProblem and MyReverseProblem). You could just use:
aima.core.search.basic.support.BasicProblem
and configure it appropriately for each usage.